### DISCUSSION TITLE
Why can't a nested container class be private?

### DISCUSSION THREAD

**Discussion by jnm2** [YairHalberstadt/stronginject#166] [Q&A]
Is there a technical reason this can't work?

```cs
using StrongInject;

partial class FooTests
{
    [Register(typeof(object))]
    private partial class TestContainer : IContainer<object> 
    {
    }
}
```

> ❌ SI0401 Module 'FooTests.TestContainer' must be public or internal.

**Comment by YairHalberstadt** [comment]
The problem is if another container imports the private one working out what it can see and what it can't becomes quite complex, so for simplicity I just allow public and internal

**Reply by jnm2** [reply]
What if I want to clearly indicate "this won't be imported as a module"?

**Reply by YairHalberstadt** [reply]
I'm happy to allow it to be used as a container but not allow it to be imported, if you want to do the work for that. It's not high priority for me, since there's a straightforward workaround.
